src/autopas/utils/AutoPasConfigurationCommunicator.h

```diff
@@ -75,6 +78,57 @@ void distributeConfigurations(std::set<ContainerOption> &containerOptions, Numbe
                               std::set<DataLayoutOption> &dataLayoutOptions, std::set<Newton3Option> &newton3Options,
                               int rank, int commSize);
 
+/**
+ * Distribute ranks in buckets, which contain only ranks with similar scenarios.
+ * Each bucket then has its own search space.
+ * @tparam Particle
+ * @param comm MPI communicator
+ * @param bucket new MPI communicator for its bucket
+ * @param container container of current simulation
+ * @param maxDifferenceForBucket maximum difference of two scenarios to get in the same bucket for MPI-tuning
+ * @param weightForMaxDensity weight for maxDensity in calculation for bucket distribution
+ */
+template <class Particle>
+void distributeRanksInBuckets(AutoPas_MPI_Comm comm, AutoPas_MPI_Comm *bucket,
+                              const std::shared_ptr<autopas::ParticleContainerInterface<Particle>> &container,
+                              double maxDifferenceForBucket, double weightForMaxDensity) {
+  int rank;
+  AutoPas_MPI_Comm_rank(comm, &rank);
+  int commSize;
+  AutoPas_MPI_Comm_size(comm, &commSize);
+
+  std::vector<double> scenarios(commSize);
+  std::pair<double, double> homogeneityAndMaxDensity =
+      autopas::utils::calculateHomogeneityAndMaxDensity<Particle>(container);
+  double scenario = homogeneityAndMaxDensity.first + weightForMaxDensity * homogeneityAndMaxDensity.second;
```

Comment:
```suggestion
  const auto [homogeneity, maxDensity] =
      autopas::utils::calculateHomogeneityAndMaxDensity<Particle>(container);
  double scenario = homogeneity + weightForMaxDensity * maxDensity;
```
rename `scenario` maybe to something like `similarityCriterion` ? Then, also rename `scenarios` accordingly.

Reply:
resolved by commit 5d64489f8fe659f14ef7b5bbc52c69f6219fb882